subprojects/core/src/main/java/org/gradle/api/internal/artifacts/configurations/ConfigurationRole.java

```diff
@@ -141,74 +102,4 @@ private static String describeDeprecation(boolean deprecated) {
             return deprecated ? " " + IS_DEPRECATED : "";
         }
     }
-
-    /**
-     * A custom implementation of {@link ConfigurationRole} that allows for non-standard combinations of usage characteristics and deprecations.
-     */
-    final class CustomConfigurationRole implements ConfigurationRole {
```

Comment:
Seems like this class could still be useful? We create anonymous implementations twice. Once in prod code and once in tests. If we pull this out to a top-level `DefaultConfigurationRole` we could reduce duplication.

Reply:
I don't really think the usage in the test should be counted towards the total - there's plenty of dummy implementations in test code.

The usage in prod code is also part of the copy functionality, which I'm going to explore deprecating next.  So I don't think this is worth doing.

Reply:
Bah.  I say this and then a very minor regression fails the merge, which, looking at this change, might be from creating that anonymous inner type.  Going to try this and see if it helps.